### PR TITLE
Move api(lgamma) from legacy_api.yaml to api.yaml

### PR DESCRIPTION
**Pull request by Charles-hit**
### PR types

Others
### PR changes

APIs
### Describe

Move api(lgamma) from legacy_api.yaml to api.yaml, and delete lgamma_op.cc,lgamma_sig.cc.

### CODE REVIEW

File: python/paddle/tensor/math.py [hunk 1]
```diff
@@ -3711,6 +3710,45 @@ def digamma(x, name=None):
...
+    r"""
+    Calculates the lgamma of the given input tensor, element-wise.
+
+    This operator performs elementwise lgamma for input $X$.
+    :math:`out = log\Gamma(x)`
+
+
+    Args:
+        x (Tensor): (Tensor), The input tensor of lgamma op.
+        with_quant_attr (BOOLEAN): Whether the operator has attributes used by quantization.
```
Comment (zyfncg) [inline python/paddle/tensor/math.py]: `with_quant_attr`不在参数中

File: python/paddle/tensor/math.py [hunk 2]
```diff
@@ -3711,6 +3710,45 @@ def digamma(x, name=None):
...
+def lgamma(x, name=None):
+    r"""
+    Calculates the lgamma of the given input tensor, element-wise.
+
+    This operator performs elementwise lgamma for input $X$.
+    :math:`out = log\Gamma(x)`
+
+
+    Args:
+        x (Tensor): (Tensor), The input tensor of lgamma op.
```
Comment (zyfncg) [inline python/paddle/tensor/math.py]: `(Tensor)`写一处即可

File: python/paddle/tensor/math.py [hunk 3]
```diff
@@ -3711,6 +3710,45 @@ def digamma(x, name=None):
...
+        with_quant_attr (BOOLEAN): Whether the operator has attributes used by quantization. 
+        name (str, optional): Name for the operation (optional, default is None). For more information, please refer to :ref:`api_guide_Name`.
+
+    Returns:
+        out (Tensor): (Tensor), The output tensor of lgamma op.
```
Comment (zyfncg) [inline python/paddle/tensor/math.py]: 同上

File: python/paddle/tensor/math.py [hunk 4]
```diff
@@ -3711,6 +3710,45 @@ def digamma(x, name=None):
...
+
+            x = paddle.to_tensor([-0.4, -0.2, 0.1, 0.3])
+            out = paddle.lgamma(x)
+            print(out)
+            # [1.31452441, 1.76149750, 2.25271273, 1.09579802]
+    """
+    if in_dygraph_mode():
+        return _C_ops.final_state_lgamma(x)
+    else:
+        if _in_legacy_dygraph():
```
Comment (zyfncg) [inline python/paddle/tensor/math.py]: 这里可以合并为`elif`

**Review by zyfncg — DISMISSED** [review]

File: python/paddle/tensor/math.py (hunk 1 above)
Comment (Charles-hit) [reply python/paddle/tensor/math.py]: done

Comment (Charles-hit) [reply python/paddle/tensor/math.py]: done

File: python/paddle/tensor/math.py (hunk 2 above)
Comment (Charles-hit) [reply python/paddle/tensor/math.py]: done

File: python/paddle/tensor/math.py (hunk 3 above)
Comment (Charles-hit) [reply python/paddle/tensor/math.py]: done

File: python/paddle/tensor/math.py (hunk 4 above)
Comment (Charles-hit) [reply python/paddle/tensor/math.py]: done

**Review by zhiboniu — APPROVED** [review]
lgtm for lgamma

**Review by zyfncg — APPROVED** [review]

**Review by Ligoml — APPROVED** [review]
LGTM for docs

**Pull request merged**